### PR TITLE
feat(manualreveal): add manual reveal scripts

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: david <david@umaproject.org>

## Motivation
https://app.shortcut.com/uma-project/story/2353/write-documentation-for-how-to-manually-reveal-via-etherscan

## Summary
This adds 2 methods for manually revealing. One is through the users signed keys pulled from the voter dapp localstorage. This is by far the easiest as it requires the least work to reveal. 

The second method requires the users backup commit seeds. It also requires the user re-enter the prices which were committed. The backup commit seeds can only be read from a file with this script.

In both cases it requires some technical knowledge to get the required information, run the scripts and execute the manual reveal, so its recommended we use this internally or allow sophisticated users to run.

Both scripts return a json string which can be pasted into etherscans `batchReveal` function on the voting contract. Documentation has been added (but no pictures) to help describe the full process.

## Testing
This has been tested against the kovan voting contract.

### CODE REVIEW

File: manualreveal/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+Click the backupCommits field, and then right click and copy the entire object. Paste this into a file inside the voter-dapp folder
+called `backupCommits`.
+
+Follow the prompts to produce the reveal string. Save the output for a manual reveal in etherscan.
+
+## Manually Revealing on Etherscan
+
+To reveal on ethereum mainnet, visit this link to [UMAs Voting Contract](https://etherscan.io/address/0x8b1631ab830d11531ae83725fda4d86012eccd77#writeContract).
+Make sure you are in the `Contract-> Write Contract` tab. Connect your Metamask wallet where it says `Connect to Web3`.
+Once connected, find the `batchReveal` function. There may be 2 of them, paste the output of the scripts into the second one.
```
Comment (daywiss) [inline manualreveal/README.md:50]: i actually dont know which function to call, they look identical

Comment (daywiss) [reply manualreveal/README.md:50]: i tested this on kovan, using the second batchReveal function and it did successfully reveal

File: manualreveal/revealWithSalt.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  const prices = await prompt.get(
+    resultsNeedingPrice.map((result, i: number) => {
+      return {
+        name: i.toString(),
+        required: true,
+        description:
+          "Enter committed price for " +
+          reveal.makeKey(result) +
+          " or 0 for No, 1 for Yes",
+      };
```
Comment (Tulun) [inline manualreveal/revealWithSalt.ts:52]: I'd use a template literal here:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: src/common/helpers/reveal.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    price:'',
+    salt:salt.toString(),
+    identifier:args.identifier,
+    time:args.time.toString(),
+    ancillaryData:args.ancillaryData,
+  }
+}
+export async function decodeEncryptedVoteWithPK(args:ethers.utils.Result,privateKey:string):Promise<PostRevealData>{
+  const {price,salt} = JSON.parse(await decryptMessage(privateKey, args.encryptedVote));
+  return {
```
Comment (Tulun) [inline src/common/helpers/reveal.ts:79]: I've often found this has errored out -- should we wrap this in a try / catch?

**Review by Tulun — APPROVED** [review]
Couple small comments overall, looks good.

**Pull request merged**